#! /bin/sh -e

# DP: capifax: add support for 3.1kHz audio

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- isdnutils-3.6.2005-01-03/capifax/capifax.c	1998-10-23 14:50:47.000000000 +0200
+++ isdnutils-3.6.2005-01-03.patched/capifax/capifax.c	2006-03-01 12:58:03.907855308 +0100
@@ -175,7 +175,7 @@
 	}
 }
 
-unsigned SendFax(char *name) {
+unsigned SendFax(char *name, unsigned int cipval) {
 	int             first;
 	char            mbuf[4];
 	unsigned	    count;
@@ -202,7 +202,7 @@
 		fclose(f);
 		return 0xFFFF;
 	}
-	Connect(&Slot, CalledPartyNumber, CallingPartyNumber, SPEECH,
+	Connect(&Slot, CalledPartyNumber, CallingPartyNumber, cipval,
 		B1PROTOCOL, B2PROTOCOL, B3PROTOCOL, (unsigned char *)&B3conf);
 	do {
 		Handle_CAPI_Msg();
@@ -254,7 +254,7 @@
 }
 
 void usage(void) {
-	fprintf(stderr, "usage: capifax [-v] [-i stationID] [-h header] [-c callerNumber] phone file\n");
+	fprintf(stderr, "usage: capifax [-a] [-v] [-i stationID] [-h header] [-c callerNumber] phone file\n");
 	exit(1);
 }
 
@@ -263,9 +263,10 @@
 	int     BChannels, Contr;
 	int     c;
 	int     ret;
+	unsigned int cipval = SPEECH;
 
 	verbose = 0;
-	while ((c = getopt(argc, argv, "vc:i:h:")) != EOF) {
+	while ((c = getopt(argc, argv, "vc:i:h:a")) != EOF) {
 		switch (c) {
 			case 'v':
 				verbose++;
@@ -279,6 +280,9 @@
 			case 'h':
 				headLine = strdup(optarg);
 				break;
+			case 'a':
+				cipval = AUDIO3_1KHZ;
+				break;
 			case '?':
 				usage();
 		}
@@ -304,7 +308,7 @@
 		fprintf(stderr, "No B-Channels available\n");
 		return -3;
 	}
-	ret = SendFax(argv[optind]);
+	ret = SendFax(argv[optind], cipval);
 	if ((Slot != INVALID_CONNECTION_ID) &&
 		(GetState(Slot) != Disconnected) &&
 		(GetState(Slot) != D_DisconnectPending))
